tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -522,19 +496,19 @@ public T WithDefault(Func<DefaultResult<T>> getDefaultValue)
             return default; // should never be invoked because we have a value for getDefaultValue
         }
 
-        public T? OverrideWith(in StructConfigurationResultWithKey<T> otelConfig)
-            => CalculateOverrides(in otelConfig, getDefaultValue: null);
+        public T? OverrideWith(in StructConfigurationResultWithKey<T> otelConfig, IConfigurationOverrideHandler overrideHandler)
+            => CalculateOverrides(overrideHandler, in otelConfig, getDefaultValue: null);
 
-        public T OverrideWith(in StructConfigurationResultWithKey<T> otelConfig, T defaultValue)
-            => CalculateOverrides(in otelConfig, getDefaultValue: () => defaultValue).Value;
+        public T OverrideWith(in StructConfigurationResultWithKey<T> otelConfig, IConfigurationOverrideHandler overrideHandler, T defaultValue)
+            => CalculateOverrides(overrideHandler, in otelConfig, getDefaultValue: () => defaultValue).Value;
 
-        public T OverrideWith(in StructConfigurationResultWithKey<T> otelConfig, Func<DefaultResult<T>> getDefaultValue)
-            => CalculateOverrides(in otelConfig, getDefaultValue).Value;
+        public T OverrideWith(in StructConfigurationResultWithKey<T> otelConfig, IConfigurationOverrideHandler overrideHandler, Func<DefaultResult<T>> getDefaultValue)
+            => CalculateOverrides(overrideHandler, in otelConfig, getDefaultValue).Value;
 
         [return: NotNullIfNotNull(nameof(getDefaultValue))]
-        private T? CalculateOverrides(in StructConfigurationResultWithKey<T> otelConfig, Func<DefaultResult<T>>? getDefaultValue)
+        private T? CalculateOverrides(IConfigurationOverrideHandler overrideHandler, in StructConfigurationResultWithKey<T> otelConfig, Func<DefaultResult<T>>? getDefaultValue)
```

Comment:
Flipping the order of the `otelConfig` and `overrideHandler` parameters here seems like an oversight, since it looks like it's `otelConfig, overrideHandler, ...` everywhere else (at least in the PR diff).